back/app/services/concerns/json_forms_ideas_overrides.rb

```diff
@@ -44,21 +44,56 @@ def custom_form_to_ui_schema fields, locale='en'
           type: 'Category',
           options: { id: 'extra' },
           label: I18n.t('custom_forms.categories.extra.title', locale: locale),
-          elements: fields.reject(&:built_in?).map{|f| yield f}
+          elements: fields.reject(&:built_in?).map{|f| yield f, previousScope = '#/properties/custom_field_values/properties/'}
         }
       ].compact)
     }
   end
 
+  def custom_form_to_json_schema(fields, locale='en')
+    {
+      type: 'object',
+      additionalProperties: false,
+      properties: fields.select(&:built_in?).inject({}) do |memo, field|
+        override_method = "#{field.resource_type.underscore}_#{field.code}_to_json_schema_field"
+        memo[field.key] =
+          if field.code && respond_to?(override_method, true)
+            send(override_method, field, locale)
+          else
+            send("#{field.input_type}_to_json_schema_field", field, locale)
+          end
+        memo.tap do |properties|
+          properties[:custom_field_values] = {
+            type: 'object',
+            additionalProperties: false,
+            properties: fields.reject(&:built_in?).each_with_object do |mem, field|
```

Comment:
There are 2 mistakes in replacing `inject` by `each_with_object`. It should be:
```suggestion
            properties: fields.reject(&:built_in?).each_with_object({}) do |field, memo|
```
Also note that the return value on line 77 is not required for `each_with_object`.